buildpack/bomfile.go

```diff
@@ -0,0 +1,160 @@
+package buildpack
+
+import (
+	"path/filepath"
+	"strings"
+
+	"github.com/pkg/errors"
+
+	"github.com/buildpacks/lifecycle/buildpack/layertypes"
+)
+
+const (
+	LayerTypeBuild LayerType = iota
+	LayerTypeCache
+	LayerTypeLaunch
+)
+
+const (
+	mediaTypeCycloneDX   = "application/vnd.cyclonedx+json"
+	mediaTypeSPDX        = "application/spdx+json"
+	mediaTypeSyft        = "application/vnd.syft+json"
+	mediaTypeUnsupported = "unsupported"
+)
+
+type LayerType int
+
+type BOMFile struct {
+	BuildpackID string
+	LayerName   string
+	LayerType   LayerType
+	Path        string
+}
+
+// Name() returns the destination filename for a given BOM file
+// cdx files should be renamed to "bom.cdx.json"
+// spdx files should be renamed to "bom.spdx.json"
+// syft files should be renamed to "bom.syft.json"
+// If the BOM is neither cdx, spdx, nor syft, the 2nd return argument
+// will return an error to indicate an unsupported format
+func (b *BOMFile) Name() (string, error) {
+	switch b.mediaType() {
+	case mediaTypeCycloneDX:
+		return "bom.cdx.json", nil
+	case mediaTypeSPDX:
+		return "bom.spdx.json", nil
+	case mediaTypeSyft:
+		return "bom.syft.json", nil
+	default:
+		return "", errors.Errorf("unsupported bom format: '%s'", b.Path)
+	}
+}
+
+func (b *BOMFile) mediaType() string {
+	name := filepath.Base(b.Path)
+
+	switch {
+	case strings.HasSuffix(name, "bom.cdx.json"):
+		return mediaTypeCycloneDX
+	case strings.HasSuffix(name, "bom.spdx.json"):
+		return mediaTypeSPDX
+	case strings.HasSuffix(name, "bom.syft.json"):
```

Comment:
```suggestion
	case strings.HasSuffix(name, ".bom.syft.json"):
```

Reply:
Addressed in https://github.com/buildpacks/lifecycle/pull/759